person.rb

```diff
@@ -27,4 +28,8 @@ def of_age?
   def can_use_services?
     @age >= 18 || @parent_permission
   end
+
+  def add_rental(book, date)
+    rentals << Rental.new(book, self, date)
+  end
```

Comment:
- [ Required ] I think it would be nice if you refactor this method for the following reason.
  
  - Since the rental object is being added to `person.rentals` in the `Rental` constructor you don't have to do it in the `add_rental` method.
  
 Please see the example code below for a suggestion of how you can achieve this.

## Example code

```ruby
require_relative 'rental'
require_relative 'nameable'

class Person < Nameable
  def add_rental(book, date)
    Rental.new(book, self, date)
  end
end
```

Reply:
Thank you again for this amazing correction too. I have implemented the changes.